src/flask_pyoidc/provider_configuration.py

```diff
@@ -164,22 +181,45 @@ def ensure_provider_metadata(self):
     def registered_client_metadata(self):
         return self._client_metadata
 
-    def register_client(self, redirect_uris, extra_parameters=None):
+    def register_client(self, client: Client, redirect_uri: str):
+
         if not self._client_metadata:
-            if 'registration_endpoint' not in self._provider_metadata:
+            if not self._provider_metadata['registration_endpoint']:
```

Comment:
This will throw a `KeyError` if `registration_endpoint` is not set.
To safe-guard against empty value this check should be `if not self._provider_metadata.get('registration_endpoint')`

Reply:
It won't. `self._provider_metadata` is an object of `ProviderMetadata`, and not a dictionary. `self._provider_metadata['registration_endpoint']` is using `__getitem__` to fetch value of the `registration_endpoint` attribute of Provider Metadata class object which is default to `None`. So if the user doesn't provide registration endpoint URL, `self._provider_metadata['registration_endpoint']` returns None. If you use `self._provider_metadata.get('registration_endpoint')`, it will throw error that `_provider_metadata` has no attribute `get`.

Reply:
The `__getitem__` directly delegates to the underlying dictionary, so I see the following:
```python
>>> from flask_pyoidc.provider_configuration import ProviderMetadata
>>> a = ProviderMetadata()
>>> b = a['registration_endpoint']
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "Flask-pyoidc/src/flask_pyoidc/provider_configuration.py", line 24, in __getitem__
    return self.store[key]
KeyError: 'registration_endpoint'
>>> b = a.get('registration_endpoint')
>>> print(b)
None
```

Reply:
I did the same. It returns `None`. I have added all default arguments in `ProviderMetadata` class so `__getitem__` will always return `None` which is default value of the argument if the `registration_endpoint` is not provided by the user. You can test your code in my branch.

```python
class ProviderMetadata(OIDCData):

    def __init__(self,
                 issuer=None,
                 authorization_endpoint=None,
                 jwks_uri=None,
                 token_endpoint=None,
                 userinfo_endpoint=None,
                 introspection_endpoint=None,
                 registration_endpoint=None,
                 **kwargs):
        """OpenID Providers have metadata describing their configuration.

        Parameters
        ----------
        issuer: str, Optional
            OP Issuer Identifier.
        authorization_endpoint: str, Optional
            URL of the OP's OAuth 2.0 Authorization Endpoint.
        jwks_uri: str, Optional
            URL of the OP's JSON Web Key Set [JWK] document.
        token_endpoint: str, Optional
            URL of the OP's OAuth 2.0 Token Endpoint.
        userinfo_endpoint: str, Optional
            URL of the OP's UserInfo Endpoint.
        introspection_endpoint: str, Optional
            URL of the OP's token introspection endpoint.
        registration_endpoint: str, Optional
            URL of the OP's Dynamic Client Registration Endpoint.
        **kwargs : dict, Optional
            Extra arguments to [OpenID Provider Metadata](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata)
        """
        super().__init__(issuer=issuer, authorization_endpoint=authorization_endpoint,
                         token_endpoint=token_endpoint, userinfo_endpoint=userinfo_endpoint,
                         jwks_uri=jwks_uri, introspection_endpoint=introspection_endpoint,
                         registration_endpoint=registration_endpoint, **kwargs)
```